api/pacman-api-asset/src/main/java/com/tmobile/pacman/api/asset/service/AssetServiceImpl.java

```diff
@@ -1285,4 +1286,21 @@ private int getCountOfNonCompliancePolicies(String assetGroup) {
             return 0;
         }
     }
+
+    public DatasourceData getAssetGroupsDataForShipperByDatasource(final String datasource) {
+        DatasourceData datasourceData = new DatasourceData();
+        Map<String, List<String>> assetGroupDomains = new HashMap<>();
+        List<String> targetTypes = repository.getTargetTypesByDatasource(datasource);
+        List<String> indices = targetTypes.stream().map(targetType ->
+                datasource + Constants.DELIMITER_UNDERSCORE + targetType).collect(Collectors.toList());
+        List<String> aliases = repository.getAliasByIndices(indices);
+        List<String> assetGroups = repository.getVisibleAssetGroupsFiltered(aliases);
+        assetGroups.forEach(assetGroup -> {
+            assetGroupDomains.put(assetGroup, getDomains(assetGroup));
+        });

```

Comment:
One more benefit from getting ALL target types in one request is that you get get those domains by yourself, because in that big response you will have already relationships between all target types (indexes) and all asset groups (aliases). So you don't need to call `getDomains` here - just do the same logic without additional requests to OS.

Let me do some math here. For AWS we have let's say 168 asset types (target types, indexes). You request aliases for each of them separately. So it is 168 requests to AWS OS. As a result you have let's say 50 user defined asset groups (aliases). And this number is not controlled by us in any way - client can define 1000 used groups easily. and we will have to deal with it.
So summary: only to collect asset group data with domains (which I have no idea what it is and it's far from to be clear from the code) 218 different requests to OS to get data which from my understanding can be gathered using single request (may be more - I didn't dive into code that much yet). I don't count DB requests yet (but I will eventually touch them some day as well).

Reply:
As for domains - I had a short chat with @plyubich  and it seems we don't care about domains anymore. It was a thing in TMobile but it means nothing for us and it needs to be gone eventually.